### ISSUE TITLE
Make CCS a module

### ISSUE THREAD

**Issue by norabelrose** [EleutherAI/elk#18]
Arguably CCS should be a `torch.nn.Module` which has the probe parameters as attributes. This also means getting rid of the NumPy stuff and using PyTorch methods throughout.

Also, the parameters like learning rate and num epochs etc. should be arguments to `fit`, not the CCS constructor.

We also might want to think about renaming the class to something like, I don't know, `ElkProbe` because:
- We should be specific that it's a probe
- Some of the changes we're considering making to the algorithm (in particular the multiclass thing #6) make it less clear that this is still _contrast_-consistent search. We should use a more generic name that encompasses all the loss functions we intend to support.

**Comment by FabienRoger** [comment]
Do you suggest merging in https://github.com/EleutherAI/elk/pull/13 before this one, or expanding it?

**Comment by norabelrose** [comment]
@FabienRoger hmmm I hadn't gotten around to looking at #13 until just now. I suppose we should expand #13 to handle this as well

**Comment by FabienRoger** [comment]
I think renaming CCS to ElkProbe should probably wait the PR with the new modular system for losses. Also it might be better since people might share takes about this name after me merge this change (I personaly like ElkProbe).
